main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -442,6 +442,10 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       SP ~ "\"" ~ zeroOrMore(!("\"" | atomic("${") | atomic("%{")) ~ Chars.CharCode) ~ "\"" ~ SP ~> ParsedAst.Literal.Str
     }
 
+    def Regex: Rule1[ParsedAst.Literal.Regex] = rule {
+      SP ~ "Regex#\"" ~ capture(zeroOrMore(!("\"") ~ CharPredicate.All)) ~ "\"" ~ SP ~> ParsedAst.Literal.Regex
```

Comment:
OK, I see that `Regex#` is consistent with our other literals. 

My initial inclination was something like `r"foo"`.

Ill explore what other languages do.

Reply:
Does the above handle escape regexs correctly? E.g. backslashes etc?

Reply:
> Does the above handle escape regexs correctly? E.g. backslashes etc?

I'd be surprised if it does. Now literals are generally working I'll add tests to check...

Reply:
The above parser code doesn't handle escapes properly - I'm investigating.

Reply:
The tests in `Test.Exp.Regex` basically check for escaping - all work.